File: app/src/main/java/com/example/schedule/SQL/SQLManager.java
package com.example.schedule.SQL;

import android.content.Context;
import android.content.SharedPreferences;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

public class SQLManager extends SQLiteOpenHelper {

    private String name;
    private Context context;

    public static final String ID = "_id";
    public static final String DAY_OF_WEEK = "day_of_week";
    public static final String COUNTER = "counter";
    public static final String SUBJECT = "subject";
    public static final String FROM = "from_";
    public static final String TO = "to_";
    public static final String AUDITORY = "auditory";
    public static final String TEACHER = "teacher";
    public static final String TYPE_OF_SUBJECT = "type_of_subject";
    public static final String BOTH_NUMERATOR_DIVIDER = "both_numerator_divider";
    public static final String SHARED_PREFERENCES_TABLES = "tables";
    public static final String SHARED_PREFERENCES_TABLE = "table";

    public static final int NUMERATOR = 1;
    public static final int DIVIDER = 2;
    public static final int BOTH = 0;

    public SQLManager(@Nullable Context context, @Nullable String name, @Nullable SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
        this.name = name;
        this.context = context;
    }

    @Override
    public void onCreate(SQLiteDatabase db) {

        if (name != null) {
            db.execSQL("CREATE TABLE IF NOT EXISTS " + name + " (" +
                    ID + " INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT," +
                    DAY_OF_WEEK + " INTEGER NOT NULL," +
                    COUNTER + " INTEGER NOT NULL," +
                    SUBJECT + " TEXT," +
                    FROM + " TEXT NOT NULL," +
                    TO + " TEXT NOT NULL," +
                    AUDITORY + " TEXT," +
                    TEACHER + " TEXT," +
                    TYPE_OF_SUBJECT + " TEXT," +
                    BOTH_NUMERATOR_DIVIDER + " INTEGER NOT NULL" +
                     ")");
            SharedPreferences sh = context.getSharedPreferences(SHARED_PREFERENCES_TABLES,Context.MODE_PRIVATE);
            sh.edit().putString(SHARED_PREFERENCES_TABLE,name).apply();
        }
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }
}

File: app/src/main/java/com/example/schedule/SQL/SQLManager.kt
package com.example.schedule.SQL

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteDatabase.CursorFactory
import android.database.sqlite.SQLiteOpenHelper

open class SQLManager(private val context: Context,
                      private val name: String,
                      factory: CursorFactory?,
                      version: Int) : SQLiteOpenHelper(context, name, factory, version) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("CREATE TABLE IF NOT EXISTS " + name + " (" +
                ID + " INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT," +
                DAY_OF_WEEK + " INTEGER NOT NULL," +
                SUBJECT + " TEXT," +
                FROM + " INTEGER NOT NULL," +
                TO + " INTEGER NOT NULL," +
                AUDITORY + " TEXT," +
                TEACHER + " TEXT," +
                TYPE_OF_SUBJECT + " TEXT" +
                ")")
        context.getSharedPreferences(SHARED_PREF_DB_TABLE_NAME, Context.MODE_PRIVATE)
                .edit()
                .putString(SHARED_PREF_TABLE_NAME_KEY, name)
                .apply()
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {}

    companion object {
        const val ID = "_id"

        //Day of week start from one
        const val DAY_OF_WEEK = "day_of_week"
        const val SUBJECT = "subject"
        const val FROM = "from_"
        const val TO = "to_"
        const val AUDITORY = "auditory"
        const val TEACHER = "teacher"
        const val TYPE_OF_SUBJECT = "STYLE_OF_SUBJECT"

        const val SHARED_PREF_DB_TABLE_NAME = "tables"
        const val SHARED_PREF_TABLE_NAME_KEY = "table"

        //For example, lecture, seminar and others
        const val VERSION = 1
    }

}
